src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeTracker.cs

```diff
@@ -185,5 +196,44 @@ int IVsFileChangeEvents.FilesChanged(uint changeCount, string[] files, uint[] ch
 
             return VSConstants.S_OK;
         }
+
+        int IVsFreeThreadedFileChangeEvents2.FilesChanged(uint cChanges, string[] rgpszFile, uint[] rggrfChange)
+        {
+            UpdatedOnDisk?.Invoke(this, EventArgs.Empty);
+
+            return VSConstants.S_OK;
+        }
+
+        int IVsFreeThreadedFileChangeEvents2.DirectoryChanged(string pszDirectory)
+        {
+            throw new Exception("We only watch files; we should never be seeing directory changes!");
+        }
+
+        int IVsFreeThreadedFileChangeEvents2.DirectoryChangedEx(string pszDirectory, string pszFile)
+        {
+            throw new Exception("We only watch files; we should never be seeing directory changes!");
+        }
+
+        int IVsFreeThreadedFileChangeEvents2.DirectoryChangedEx2(string pszDirectory, uint cChanges, string[] rgpszFile, uint[] rggrfChange)
+        {
+            throw new Exception("We only watch files; we should never be seeing directory changes!");
+        }
+
+        int IVsFreeThreadedFileChangeEvents.FilesChanged(uint cChanges, string[] rgpszFile, uint[] rggrfChange)
+        {
+            UpdatedOnDisk?.Invoke(this, EventArgs.Empty);
```

Comment:
intresting.  so we just lose which files actually changed?

Reply:
This is the legacy watcher wrapper we have that just only listens to a single file ever.